flash/text/classification/data.py

```diff
@@ -11,247 +11,168 @@
+    def _resolve_target(target_keys: Union[str, List[str]], element: Dict[str, Any]) -> Dict[str, Any]:
+        if not isinstance(target_keys, List):
+            element[DataKeys.TARGET] = element.pop(target_keys)
+        else:
+            element[DataKeys.TARGET] = [element[target_key] for target_key in target_keys]
         return element
 
-    def _to_hf_dataset(self, data) -> Sequence[Mapping[str, Any]]:
-        """account for flash CI testing context."""
-        hf_dataset, *other = self.to_hf_dataset(data)
-
-        if flash._IS_TESTING and not torch.cuda.is_available():
-            # NOTE: must subset in this way to return a Dataset
-            hf_dataset = hf_dataset.select(range(20))
-
-        return (hf_dataset, *other)
-
+    @requires("text")
     def load_data(
         self,
-        data: Tuple[str, Union[str, List[str]], Union[str, List[str]]],
-        dataset: Optional[Any] = None,
-    ) -> Sequence[Mapping[str, Any]]:
+        hf_dataset: Dataset,
+        input_key: str,
+        target_keys: Optional[Union[str, List[str]]] = None,
+        max_length: int = 128,
+    ) -> Dataset:
         """Loads data into HuggingFace datasets.Dataset."""
-
-        hf_dataset, input, *other = self._to_hf_dataset(data)
+        self.max_length = max_length
 
         if not self.predicting:
-            target: Union[str, List[str]] = other.pop()
-            if isinstance(target, List):
-                # multi-target
-                dataset.multi_label = True
-                hf_dataset = hf_dataset.map(partial(self._multilabel_target, target))  # NOTE: renames target column
-                dataset.num_classes = len(target)
-                self.set_state(ClassificationState(target))
-            else:
-                dataset.multi_label = False
-                if self.training:
-                    labels = list(sorted(list(set(hf_dataset[target]))))
-                    dataset.num_classes = len(labels)
-                    self.set_state(ClassificationState(labels))
-
-                labels = self.get_state(ClassificationState)
-
-                # convert labels to ids (note: the target column get overwritten)
-                if labels is not None:
-                    labels = labels.labels
-                    label_to_class_mapping = {v: k for k, v in enumerate(labels)}
-                    hf_dataset = hf_dataset.map(partial(self._transform_label, label_to_class_mapping, target))
-
-                # rename label column
-                hf_dataset = hf_dataset.rename_column(target, DataKeys.TARGET)
+            hf_dataset = hf_dataset.map(partial(self._resolve_target, target_keys))
+            targets = hf_dataset.to_dict()[DataKeys.TARGET]
+            self.load_target_metadata(targets)
+
+            # If we had binary multi-class targets then we also know the labels (column names)
+            if self.target_mode is TargetMode.MULTI_BINARY and isinstance(target_keys, List):
```

Comment:
Why do you override the state there ? If the state is shared among train, val. This would override the train one.

Reply:
The classification state is only ever shared from the train data (becuase labels are always inferred from the train data rather than val. or test). In this case, the labels are binary, so we assume that a one in a column corresponds to having that label. So the labels inferred by our classification utils are `None` but we replace them with the target keys from the dataset.